File: binding/src/libqcborarray.cpp
#include <QCborArray>
#define WORKAROUND_INNER_CLASS_DEFINITION_QCborArray__ConstIterator
#define WORKAROUND_INNER_CLASS_DEFINITION_QCborArray__Iterator
#include <QCborValue>
#include <QCborValueConstRef>
#include <QCborValueRef>
#include <QChar>
#include <QDataStream>
#include <QJsonArray>
#include <qcborarray.h>
#include "libqcborarray.h"
#include "libqcborarray.hxx"

QCborArray* QCborArray_new() {
	 return new QCborArray();
}

QCborArray* QCborArray_new2(const QCborArray* other) {
	 return new QCborArray(*other);
}

void QCborArray_OperatorAssign(QCborArray* self, const QCborArray* other) {
	self->operator=(*other);
}

void QCborArray_Swap(QCborArray* self, QCborArray* other) {
	self->swap(*other);
}

QCborValue* QCborArray_ToCborValue(const QCborArray* self) {
	return new QCborValue(self->toCborValue());
}

ptrdiff_t QCborArray_Size(const QCborArray* self) {
	return self->size();
}

bool QCborArray_IsEmpty(const QCborArray* self) {
	return self->isEmpty();
}

void QCborArray_Clear(QCborArray* self) {
	self->clear();
}

QCborValue* QCborArray_At(const QCborArray* self, ptrdiff_t i) {
	return new QCborValue(self->at(i));
}

QCborValue* QCborArray_First(const QCborArray* self) {
	return new QCborValue(self->first());
}

QCborValue* QCborArray_Last(const QCborArray* self) {
	return new QCborValue(self->last());
}

QCborValue* QCborArray_OperatorSubscript(const QCborArray* self, ptrdiff_t i) {
	return new QCborValue(self->operator[](i));
}

QCborValueRef* QCborArray_First2(QCborArray* self) {
	return new QCborValueRef(self->first());
}

QCborValueRef* QCborArray_Last2(QCborArray* self) {
	return new QCborValueRef(self->last());
}

QCborValueRef* QCborArray_OperatorSubscript2(QCborArray* self, ptrdiff_t i) {
	return new QCborValueRef(self->operator[](i));
}

void QCborArray_Insert(QCborArray* self, ptrdiff_t i, const QCborValue* value) {
	self->insert(i, *value);
}

void QCborArray_Prepend(QCborArray* self, const QCborValue* value) {
	self->prepend(*value);
}

void QCborArray_Append(QCborArray* self, const QCborValue* value) {
	self->append(*value);
}

QCborValue* QCborArray_Extract(QCborArray* self, const QChar* it) {
	return new QCborValue(self->extract(it));
}

QCborValue* QCborArray_Extract2(QCborArray* self, QTreeWidgetItemIterator::Iterator it) {
	return new QCborValue(self->extract(it));
}

void QCborArray_RemoveAt(QCborArray* self, ptrdiff_t i) {
	self->removeAt(i);
}

QCborValue* QCborArray_TakeAt(QCborArray* self, ptrdiff_t i) {
	return new QCborValue(self->takeAt(i));
}

void QCborArray_RemoveFirst(QCborArray* self) {
	self->removeFirst();
}

void QCborArray_RemoveLast(QCborArray* self) {
	self->removeLast();
}

QCborValue* QCborArray_TakeFirst(QCborArray* self) {
	return new QCborValue(self->takeFirst());
}

QCborValue* QCborArray_TakeLast(QCborArray* self) {
	return new QCborValue(self->takeLast());
}

bool QCborArray_Contains(const QCborArray* self, const QCborValue* value) {
	return self->contains(*value);
}

int QCborArray_Compare(const QCborArray* self, const QCborArray* other) {
	return self->compare(*other);
}

QCborArray__Iterator* QCborArray_Begin(QCborArray* self) {
	return new QCborArray::Iterator(self->begin());
}

QCborArray__ConstIterator* QCborArray_ConstBegin(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->constBegin());
}

QCborArray__ConstIterator* QCborArray_Begin2(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->begin());
}

QCborArray__ConstIterator* QCborArray_Cbegin(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->cbegin());
}

QCborArray__Iterator* QCborArray_End(QCborArray* self) {
	return new QCborArray::Iterator(self->end());
}

QCborArray__ConstIterator* QCborArray_ConstEnd(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->constEnd());
}

QCborArray__ConstIterator* QCborArray_End2(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->end());
}

QCborArray__ConstIterator* QCborArray_Cend(const QCborArray* self) {
	return new QCborArray::ConstIterator(self->cend());
}

QCborArray__Iterator* QCborArray_Insert2(QCborArray* self, QCborArray__Iterator* before, const QCborValue* value) {
	return new QCborArray::Iterator(self->insert(*before, *value));
}

QCborArray__Iterator* QCborArray_Insert3(QCborArray* self, QCborArray__ConstIterator* before, const QCborValue* value) {
	return new QCborArray::Iterator(self->insert(*before, *value));
}

QCborArray__Iterator* QCborArray_Erase(QCborArray* self, QCborArray__Iterator* it) {
	return new QCborArray::Iterator(self->erase(*it));
}

QCborArray__Iterator* QCborArray_Erase2(QCborArray* self, QCborArray__ConstIterator* it) {
	return new QCborArray::Iterator(self->erase(*it));
}

void QCborArray_PushBack(QCborArray* self, const QCborValue* t) {
	self->push_back(*t);
}

void QCborArray_PushFront(QCborArray* self, const QCborValue* t) {
	self->push_front(*t);
}

void QCborArray_PopFront(QCborArray* self) {
	self->pop_front();
}

void QCborArray_PopBack(QCborArray* self) {
	self->pop_back();
}

bool QCborArray_Empty(const QCborArray* self) {
	return self->empty();
}

QCborArray* QCborArray_OperatorPlus(const QCborArray* self, const QCborValue* v) {
	return new QCborArray(self->operator+(*v));
}

QCborArray* QCborArray_OperatorPlusAssign(QCborArray* self, const QCborValue* v) {
	return new QCborArray(self->operator+=(*v));
}

QCborArray* QCborArray_OperatorShiftLeft(QCborArray* self, const QCborValue* v) {
	return new QCborArray(self->operator<<(*v));
}

QCborArray* QCborArray_FromStringList(const libqt_list list) {
	return new QCborArray(QCborArray::fromStringList(QList<QString>()));
}

QCborArray* QCborArray_FromVariantList(const libqt_list list) {
	return new QCborArray(QCborArray::fromVariantList(QList<QVariant>()));
}

QCborArray* QCborArray_FromJsonArray(const QJsonArray* array) {
	return new QCborArray(QCborArray::fromJsonArray(*array));
}

libqt_list QCborArray_ToVariantList(const QCborArray* self) {
	auto _ret = self->toVariantList();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		auto& _elem = _ret[_i];
		_data[_i] = new std::remove_reference_t<decltype(_elem)>(_elem);
	}
	return _arr;
}

QJsonArray* QCborArray_ToJsonArray(const QCborArray* self) {
	return new QJsonArray(self->toJsonArray());
}

void QCborArray_Delete(QCborArray* self) {
    delete self;
}

QCborArray__Iterator* QCborArray__Iterator_new(const QCborArray__Iterator* other) {
	 return new QCborArray::Iterator(*other);
}

QCborArray__Iterator* QCborArray__Iterator_new2() {
	 return new QCborArray::Iterator();
}

QCborArray__Iterator* QCborArray__Iterator_new3(const QCborArray__Iterator* param1) {
	 return new QCborArray::Iterator(*param1);
}

void QCborArray__Iterator_OperatorAssign(QCborArray__Iterator* self, const QCborArray__Iterator* other) {
	self->operator=(*other);
}

QCborValueRef* QCborArray__Iterator_OperatorMultiply(const QCborArray__Iterator* self) {
	return new QCborValueRef(self->operator*());
}

QCborValueRef* QCborArray__Iterator_OperatorMinusGreater(QCborArray__Iterator* self) {
	return self->operator->();
}

const QCborValueConstRef* QCborArray__Iterator_OperatorMinusGreater2(const QCborArray__Iterator* self) {
	return self->operator->();
}

QCborValueRef* QCborArray__Iterator_OperatorSubscript(const QCborArray__Iterator* self, ptrdiff_t j) {
	return new QCborValueRef(self->operator[](j));
}

QTreeWidgetItemIterator::Iterator* QCborArray__Iterator_OperatorPlusPlus(QCborArray__Iterator* self) {
	return self->operator++();
}

QTreeWidgetItemIterator::Iterator QCborArray__Iterator_OperatorPlusPlus2(QCborArray__Iterator* self, int param1) {
	return self->operator++(param1);
}

QTreeWidgetItemIterator::Iterator* QCborArray__Iterator_OperatorMinusMinus(QCborArray__Iterator* self) {
	return self->operator--();
}

QTreeWidgetItemIterator::Iterator QCborArray__Iterator_OperatorMinusMinus2(QCborArray__Iterator* self, int param1) {
	return self->operator--(param1);
}

QTreeWidgetItemIterator::Iterator* QCborArray__Iterator_OperatorPlusAssign(QCborArray__Iterator* self, ptrdiff_t j) {
	return self->operator+=(j);
}

QTreeWidgetItemIterator::Iterator* QCborArray__Iterator_OperatorMinusAssign(QCborArray__Iterator* self, ptrdiff_t j) {
	return self->operator-=(j);
}

QTreeWidgetItemIterator::Iterator QCborArray__Iterator_OperatorPlus(const QCborArray__Iterator* self, ptrdiff_t j) {
	return self->operator+(j);
}

QTreeWidgetItemIterator::Iterator QCborArray__Iterator_OperatorMinus(const QCborArray__Iterator* self, ptrdiff_t j) {
	return self->operator-(j);
}

ptrdiff_t QCborArray__Iterator_OperatorMinus2(const QCborArray__Iterator* self, QCborArray__Iterator* j) {
	return self->operator-(*j);
}

void QCborArray__Iterator_Delete(QCborArray__Iterator* self) {
    delete self;
}

QCborArray__ConstIterator* QCborArray__ConstIterator_new(const QCborArray__ConstIterator* other) {
	 return new QCborArray::ConstIterator(*other);
}

QCborArray__ConstIterator* QCborArray__ConstIterator_new2() {
	 return new QCborArray::ConstIterator();
}

QCborArray__ConstIterator* QCborArray__ConstIterator_new3(const QCborArray__ConstIterator* param1) {
	 return new QCborArray::ConstIterator(*param1);
}

void QCborArray__ConstIterator_OperatorAssign(QCborArray__ConstIterator* self, const QCborArray__ConstIterator* other) {
	self->operator=(*other);
}

QCborValueConstRef* QCborArray__ConstIterator_OperatorMultiply(const QCborArray__ConstIterator* self) {
	return new QCborValueConstRef(self->operator*());
}

const QCborValueConstRef* QCborArray__ConstIterator_OperatorMinusGreater(const QCborArray__ConstIterator* self) {
	return self->operator->();
}

QCborValueConstRef* QCborArray__ConstIterator_OperatorSubscript(const QCborArray__ConstIterator* self, ptrdiff_t j) {
	return new QCborValueConstRef(self->operator[](j));
}

const QChar* QCborArray__ConstIterator_OperatorPlusPlus(QCborArray__ConstIterator* self) {
	ConstIterator& _ret = self->operator++();
	// Cast returned reference into pointer
	return const_cast<QChar*>(&_ret);
}

const QChar* QCborArray__ConstIterator_OperatorPlusPlus2(QCborArray__ConstIterator* self, int param1) {
	return self->operator++(param1);
}

const QChar* QCborArray__ConstIterator_OperatorMinusMinus(QCborArray__ConstIterator* self) {
	ConstIterator& _ret = self->operator--();
	// Cast returned reference into pointer
	return const_cast<QChar*>(&_ret);
}

const QChar* QCborArray__ConstIterator_OperatorMinusMinus2(QCborArray__ConstIterator* self, int param1) {
	return self->operator--(param1);
}

const QChar* QCborArray__ConstIterator_OperatorPlusAssign(QCborArray__ConstIterator* self, ptrdiff_t j) {
	ConstIterator& _ret = self->operator+=(j);
	// Cast returned reference into pointer
	return const_cast<QChar*>(&_ret);
}

const QChar* QCborArray__ConstIterator_OperatorMinusAssign(QCborArray__ConstIterator* self, ptrdiff_t j) {
	ConstIterator& _ret = self->operator-=(j);
	// Cast returned reference into pointer
	return const_cast<QChar*>(&_ret);
}

const QChar* QCborArray__ConstIterator_OperatorPlus(const QCborArray__ConstIterator* self, ptrdiff_t j) {
	return self->operator+(j);
}

const QChar* QCborArray__ConstIterator_OperatorMinus(const QCborArray__ConstIterator* self, ptrdiff_t j) {
	return self->operator-(j);
}

ptrdiff_t QCborArray__ConstIterator_OperatorMinus2(const QCborArray__ConstIterator* self, QCborArray__ConstIterator* j) {
	return self->operator-(*j);
}

void QCborArray__ConstIterator_Delete(QCborArray__ConstIterator* self) {
    delete self;
}

File: binding/src/libqcborarray.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcborarray.html
pub const qcborarray = struct {

    /// New constructs a new QCborArray object.
    pub fn New() QtC.QCborArray {
        return qtc.QCborArray_new();
    }


    /// New2 constructs a new QCborArray object.
    pub fn New2(other: ?*anyopaque) QtC.QCborArray {
        return qtc.QCborArray_new2(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCborArray_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCborArray_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn ToCborValue(self: ?*anyopaque, ) QtC.QCborValue {
        return qtc.QCborArray_ToCborValue(@ptrCast(self));
    }

    pub fn Size(self: ?*anyopaque, ) i64 {
        return qtc.QCborArray_Size(@ptrCast(self));
    }

    pub fn IsEmpty(self: ?*anyopaque, ) bool {
        return qtc.QCborArray_IsEmpty(@ptrCast(self));
    }

    pub fn Clear(self: ?*anyopaque, ) void {
        qtc.QCborArray_Clear(@ptrCast(self));
    }

    pub fn At(self: ?*anyopaque, i: i64) QtC.QCborValue {
        return qtc.QCborArray_At(@ptrCast(self), i);
    }

    pub fn First(self: ?*anyopaque, ) QtC.QCborValue {
        return qtc.QCborArray_First(@ptrCast(self));
    }

    pub fn Last(self: ?*anyopaque, ) QtC.QCborValue {
        return qtc.QCborArray_Last(@ptrCast(self));
    }

    pub fn OperatorSubscript(self: ?*anyopaque, i: i64) QtC.QCborValue {
        return qtc.QCborArray_OperatorSubscript(@ptrCast(self), i);
    }

    pub fn First2(self: ?*anyopaque, ) QtC.QCborValueRef {
        return qtc.QCborArray_First2(@ptrCast(self));
    }

    pub fn Last2(self: ?*anyopaque, ) QtC.QCborValueRef {
        return qtc.QCborArray_Last2(@ptrCast(self));
    }

    pub fn OperatorSubscript2(self: ?*anyopaque, i: i64) QtC.QCborValueRef {
        return qtc.QCborArray_OperatorSubscript2(@ptrCast(self), i);
    }

    pub fn Insert(self: ?*anyopaque, i: i64, value: ?*anyopaque) void {
        qtc.QCborArray_Insert(@ptrCast(self), i, @ptrCast(value));
    }

    pub fn Prepend(self: ?*anyopaque, value: ?*anyopaque) void {
        qtc.QCborArray_Prepend(@ptrCast(self), @ptrCast(value));
    }

    pub fn Append(self: ?*anyopaque, value: ?*anyopaque) void {
        qtc.QCborArray_Append(@ptrCast(self), @ptrCast(value));
    }

    pub fn Extract(self: ?*anyopaque, it: ?*anyopaque) QtC.QCborValue {
        return qtc.QCborArray_Extract(@ptrCast(self), @ptrCast(it));
    }

    pub fn Extract2(self: ?*anyopaque, it: QTreeWidgetItemIterator::Iterator) QtC.QCborValue {
        return qtc.QCborArray_Extract2(@ptrCast(self), it);
    }

    pub fn RemoveAt(self: ?*anyopaque, i: i64) void {
        qtc.QCborArray_RemoveAt(@ptrCast(self), i);
    }

    pub fn TakeAt(self: ?*anyopaque, i: i64) QtC.QCborValue {
        return qtc.QCborArray_TakeAt(@ptrCast(self), i);
    }

    pub fn RemoveFirst(self: ?*anyopaque, ) void {
        qtc.QCborArray_RemoveFirst(@ptrCast(self));
    }

    pub fn RemoveLast(self: ?*anyopaque, ) void {
        qtc.QCborArray_RemoveLast(@ptrCast(self));
    }

    pub fn TakeFirst(self: ?*anyopaque, ) QtC.QCborValue {
        return qtc.QCborArray_TakeFirst(@ptrCast(self));
    }

    pub fn TakeLast(self: ?*anyopaque, ) QtC.QCborValue {
        return qtc.QCborArray_TakeLast(@ptrCast(self));
    }

    pub fn Contains(self: ?*anyopaque, value: ?*anyopaque) bool {
        return qtc.QCborArray_Contains(@ptrCast(self), @ptrCast(value));
    }

    pub fn Compare(self: ?*anyopaque, other: ?*anyopaque) i32 {
        return qtc.QCborArray_Compare(@ptrCast(self), @ptrCast(other));
    }

    pub fn Begin(self: ?*anyopaque, ) QtC.QCborArray__Iterator {
        return qtc.QCborArray_Begin(@ptrCast(self));
    }

    pub fn ConstBegin(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_ConstBegin(@ptrCast(self));
    }

    pub fn Begin2(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_Begin2(@ptrCast(self));
    }

    pub fn Cbegin(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_Cbegin(@ptrCast(self));
    }

    pub fn End(self: ?*anyopaque, ) QtC.QCborArray__Iterator {
        return qtc.QCborArray_End(@ptrCast(self));
    }

    pub fn ConstEnd(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_ConstEnd(@ptrCast(self));
    }

    pub fn End2(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_End2(@ptrCast(self));
    }

    pub fn Cend(self: ?*anyopaque, ) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray_Cend(@ptrCast(self));
    }

    pub fn Insert2(self: ?*anyopaque, before: QtC.QCborArray__Iterator, value: ?*anyopaque) QtC.QCborArray__Iterator {
        return qtc.QCborArray_Insert2(@ptrCast(self), @ptrCast(before), @ptrCast(value));
    }

    pub fn Insert3(self: ?*anyopaque, before: QtC.QCborArray__ConstIterator, value: ?*anyopaque) QtC.QCborArray__Iterator {
        return qtc.QCborArray_Insert3(@ptrCast(self), @ptrCast(before), @ptrCast(value));
    }

    pub fn Erase(self: ?*anyopaque, it: QtC.QCborArray__Iterator) QtC.QCborArray__Iterator {
        return qtc.QCborArray_Erase(@ptrCast(self), @ptrCast(it));
    }

    pub fn Erase2(self: ?*anyopaque, it: QtC.QCborArray__ConstIterator) QtC.QCborArray__Iterator {
        return qtc.QCborArray_Erase2(@ptrCast(self), @ptrCast(it));
    }

    pub fn PushBack(self: ?*anyopaque, t: ?*anyopaque) void {
        qtc.QCborArray_PushBack(@ptrCast(self), @ptrCast(t));
    }

    pub fn PushFront(self: ?*anyopaque, t: ?*anyopaque) void {
        qtc.QCborArray_PushFront(@ptrCast(self), @ptrCast(t));
    }

    pub fn PopFront(self: ?*anyopaque, ) void {
        qtc.QCborArray_PopFront(@ptrCast(self));
    }

    pub fn PopBack(self: ?*anyopaque, ) void {
        qtc.QCborArray_PopBack(@ptrCast(self));
    }

    pub fn Empty(self: ?*anyopaque, ) bool {
        return qtc.QCborArray_Empty(@ptrCast(self));
    }

    pub fn OperatorPlus(self: ?*anyopaque, v: ?*anyopaque) QtC.QCborArray {
        return qtc.QCborArray_OperatorPlus(@ptrCast(self), @ptrCast(v));
    }

    pub fn OperatorPlusAssign(self: ?*anyopaque, v: ?*anyopaque) QtC.QCborArray {
        return qtc.QCborArray_OperatorPlusAssign(@ptrCast(self), @ptrCast(v));
    }

    pub fn OperatorShiftLeft(self: ?*anyopaque, v: ?*anyopaque) QtC.QCborArray {
        return qtc.QCborArray_OperatorShiftLeft(@ptrCast(self), @ptrCast(v));
    }

    pub fn FromStringList(list: []const u8) QtC.QCborArray {
        return qtc.QCborArray_FromStringList(@ptrCast(list));
    }

    pub fn FromVariantList(list: []const u8) QtC.QCborArray {
        return qtc.QCborArray_FromVariantList(@ptrCast(list));
    }

    pub fn FromJsonArray(array: ?*anyopaque) QtC.QCborArray {
        return qtc.QCborArray_FromJsonArray(@ptrCast(array));
    }

    pub fn ToVariantList(self: ?*anyopaque, ) []const u8 {
        return qtc.QCborArray_ToVariantList(@ptrCast(self));
    }

    pub fn ToJsonArray(self: ?*anyopaque, ) QtC.QJsonArray {
        return qtc.QCborArray_ToJsonArray(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCborArray_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qcborarray-iterator.html
pub const qcborarray__iterator = struct {

    /// New constructs a new QCborArray::Iterator object.
    pub fn New(other: ?*anyopaque) QtC.QCborArray__Iterator {
        return qtc.QCborArray__Iterator_new(@ptrCast(other));
    }


    /// New2 constructs a new QCborArray::Iterator object.
    pub fn New2() QtC.QCborArray__Iterator {
        return qtc.QCborArray__Iterator_new2();
    }


    /// New3 constructs a new QCborArray::Iterator object.
    pub fn New3(param1: ?*anyopaque) QtC.QCborArray__Iterator {
        return qtc.QCborArray__Iterator_new3(@ptrCast(param1));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCborArray__Iterator_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorMultiply(self: ?*anyopaque, ) QtC.QCborValueRef {
        return qtc.QCborArray__Iterator_OperatorMultiply(@ptrCast(self));
    }

    pub fn OperatorMinusGreater(self: ?*anyopaque, ) QtC.QCborValueRef {
        return qtc.QCborArray__Iterator_OperatorMinusGreater(@ptrCast(self));
    }

    pub fn OperatorMinusGreater2(self: ?*anyopaque, ) QtC.QCborValueConstRef {
        return qtc.QCborArray__Iterator_OperatorMinusGreater2(@ptrCast(self));
    }

    pub fn OperatorSubscript(self: ?*anyopaque, j: i64) QtC.QCborValueRef {
        return qtc.QCborArray__Iterator_OperatorSubscript(@ptrCast(self), j);
    }

    pub fn OperatorPlusPlus(self: ?*anyopaque, ) QTreeWidgetItemIterator::Iterator {
        return @ptrCast(qtc.QCborArray__Iterator_OperatorPlusPlus(@ptrCast(self)));
    }

    pub fn OperatorPlusPlus2(self: ?*anyopaque, param1: i32) QTreeWidgetItemIterator::Iterator {
        return qtc.QCborArray__Iterator_OperatorPlusPlus2(@ptrCast(self), param1);
    }

    pub fn OperatorMinusMinus(self: ?*anyopaque, ) QTreeWidgetItemIterator::Iterator {
        return @ptrCast(qtc.QCborArray__Iterator_OperatorMinusMinus(@ptrCast(self)));
    }

    pub fn OperatorMinusMinus2(self: ?*anyopaque, param1: i32) QTreeWidgetItemIterator::Iterator {
        return qtc.QCborArray__Iterator_OperatorMinusMinus2(@ptrCast(self), param1);
    }

    pub fn OperatorPlusAssign(self: ?*anyopaque, j: i64) QTreeWidgetItemIterator::Iterator {
        return @ptrCast(qtc.QCborArray__Iterator_OperatorPlusAssign(@ptrCast(self), j));
    }

    pub fn OperatorMinusAssign(self: ?*anyopaque, j: i64) QTreeWidgetItemIterator::Iterator {
        return @ptrCast(qtc.QCborArray__Iterator_OperatorMinusAssign(@ptrCast(self), j));
    }

    pub fn OperatorPlus(self: ?*anyopaque, j: i64) QTreeWidgetItemIterator::Iterator {
        return qtc.QCborArray__Iterator_OperatorPlus(@ptrCast(self), j);
    }

    pub fn OperatorMinus(self: ?*anyopaque, j: i64) QTreeWidgetItemIterator::Iterator {
        return qtc.QCborArray__Iterator_OperatorMinus(@ptrCast(self), j);
    }

    pub fn OperatorMinus2(self: ?*anyopaque, j: QtC.QCborArray__Iterator) i64 {
        return qtc.QCborArray__Iterator_OperatorMinus2(@ptrCast(self), @ptrCast(j));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCborArray__Iterator_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qcborarray-constiterator.html
pub const qcborarray__constiterator = struct {

    /// New constructs a new QCborArray::ConstIterator object.
    pub fn New(other: ?*anyopaque) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray__ConstIterator_new(@ptrCast(other));
    }


    /// New2 constructs a new QCborArray::ConstIterator object.
    pub fn New2() QtC.QCborArray__ConstIterator {
        return qtc.QCborArray__ConstIterator_new2();
    }


    /// New3 constructs a new QCborArray::ConstIterator object.
    pub fn New3(param1: ?*anyopaque) QtC.QCborArray__ConstIterator {
        return qtc.QCborArray__ConstIterator_new3(@ptrCast(param1));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCborArray__ConstIterator_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorMultiply(self: ?*anyopaque, ) QtC.QCborValueConstRef {
        return qtc.QCborArray__ConstIterator_OperatorMultiply(@ptrCast(self));
    }

    pub fn OperatorMinusGreater(self: ?*anyopaque, ) QtC.QCborValueConstRef {
        return qtc.QCborArray__ConstIterator_OperatorMinusGreater(@ptrCast(self));
    }

    pub fn OperatorSubscript(self: ?*anyopaque, j: i64) QtC.QCborValueConstRef {
        return qtc.QCborArray__ConstIterator_OperatorSubscript(@ptrCast(self), j);
    }

    pub fn OperatorPlusPlus(self: ?*anyopaque, ) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorPlusPlus(@ptrCast(self));
    }

    pub fn OperatorPlusPlus2(self: ?*anyopaque, param1: i32) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorPlusPlus2(@ptrCast(self), param1);
    }

    pub fn OperatorMinusMinus(self: ?*anyopaque, ) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorMinusMinus(@ptrCast(self));
    }

    pub fn OperatorMinusMinus2(self: ?*anyopaque, param1: i32) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorMinusMinus2(@ptrCast(self), param1);
    }

    pub fn OperatorPlusAssign(self: ?*anyopaque, j: i64) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorPlusAssign(@ptrCast(self), j);
    }

    pub fn OperatorMinusAssign(self: ?*anyopaque, j: i64) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorMinusAssign(@ptrCast(self), j);
    }

    pub fn OperatorPlus(self: ?*anyopaque, j: i64) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorPlus(@ptrCast(self), j);
    }

    pub fn OperatorMinus(self: ?*anyopaque, j: i64) QtC.QChar {
        return qtc.QCborArray__ConstIterator_OperatorMinus(@ptrCast(self), j);
    }

    pub fn OperatorMinus2(self: ?*anyopaque, j: QtC.QCborArray__ConstIterator) i64 {
        return qtc.QCborArray__ConstIterator_OperatorMinus2(@ptrCast(self), @ptrCast(j));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCborArray__ConstIterator_Delete(@ptrCast(self));
    }
};
